server/index.js

```diff
@@ -1,14 +1,36 @@
 require('dotenv/config');
 const express = require('express');
 const staticMiddleware = require('./static-middleware');
+const ClientError = require('./client-error');
 const errorMiddleware = require('./error-middleware');
+const axios = require('axios');
+const crypto = require('node:crypto');
 
 const app = express();
 
+const API_PUBLIC_KEY = 'b8483fd7fba99cd20a9fefc4e5106f88';
+const API_PRIVATE_KEY = 'c2746ff73c66112e104538fe16622cbb21205d8f';
+
 app.use(staticMiddleware);
 
-app.get('/api/hello', (req, res) => {
-  res.json({ hello: 'world' });
+app.get('/marvel/:characterName', (req, res) => {
+
+  const timestamp = Date.now().toString();
+  const hash = crypto.createHash('md5').update(timestamp + API_PRIVATE_KEY + API_PUBLIC_KEY).digest('hex');
+  const characterName = req.params.characterName;
+  const url = `https://gateway.marvel.com/v1/public/characters?apikey=${API_PUBLIC_KEY}&ts=${timestamp}&hash=${hash}&name=${encodeURIComponent(characterName)}`;
+
+  axios.get(url)
+    .then(({ data: { data: { results } } }) => {
+      const { name, description = 'None Available', thumbnail, comics } = results[0] || {};
+      const characterThumbnailUrl = thumbnail ? `${thumbnail.path}.${thumbnail.extension}` : 'None Available';
+      const characterComicAppearances = comics ? comics.available : 'None Available';
+      res.status(200).json(`Name: ${name}, Description: ${description || 'None Available'}, Thumbnail URL: ${characterThumbnailUrl}, Comic Appearances: ${characterComicAppearances}`);
+    })
+    .catch(error => {
+      throw new ClientError(500, error);
```

Comment:
I think this will work, but you should force an error and try it out. We haven't done much with throwing exceptions out of Promise.catch.